src/basic/random-util.c

```diff
@@ -494,3 +494,22 @@ int random_write_entropy(int fd, const void *seed, size_t size, bool credit) {
 
         return 1;
 }
+
+int random_u64_range(uint64_t m) {
+        uint64_t x, remainder;
+
+        /* Generates a random number in the range 0…m-1, unbiased. (Java's algorithm) */
+
+        if (m == 0) /* Let's take m == 0 as special case to return an integer from the full range */
+                return random_u64();
+        if (m == 1)
+                return 0;
+
+        remainder = UINT64_MAX % m;
+
+        do {
+                x = random_u64();
```

Comment:
@poettering @keszybz coverity complains about this line:

> CID 1446662:  Insecure data handling  (TAINTED_SCALAR)
> Using tainted variable "x" as a loop boundary.

Any clue what "tainted" means?

Reply:
I figure it detected that x might be initialized from a read(), and is used without validation. It's of course misleading here, since the source is trusted (/dev/urandom, i.e the kernel), and it's safe as long as the data is actually resonably random.

Tracking "taintedness" certainly makes sense for reading arbitrary files, but not here.